prober/http.go

```diff
@@ -217,7 +217,20 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 		// the hostname of the target.
 		httpClientConfig.TLSConfig.ServerName = targetHost
 	}
-	client, err := pconfig.NewHTTPClientFromConfig(&httpClientConfig)
+
+	// Target protocol is used as hint for local address protocol.
+	sourceNetwork := "ip4"
+	if ip.IP.To4() == nil {
+		sourceNetwork = "ip6"
+	}
+	// Resolve local bind address
+	saddr, err := net.ResolveIPAddr(sourceNetwork, module.SourceAddress)
```

Comment:
What if there is no source address configured?

Reply:
Condition is indeed missing here. Will update soon.